### PR TITLE
Copter : implement initial mode

### PR DESCRIPTION
**Pull request by khancyr**
This implement initial mode support like in other vehicle.
This allow to set an initial mode in case of no radio. Default behavior is not change, if a radio is present the mode switch will have the last word on init.
In case we cannot set the initial mode at start, we will backup on STABILIZE like currently.

To test this on SITL, we need to set : 
param set SIM_RC_FAIL 1
param set INITIAL_MODE 9

PR :

```
BUILD SUMMARY
Build directory: /home/khancyr/Workspace/ardupilot/build/sitl
Target          Text     Data    BSS     Total  
------------------------------------------------
bin/arducopter  3143941  131715  110624  3386280
```

MASTER : 

```
BUILD SUMMARY
Build directory: /home/khancyr/Workspace/ardupilot/build/sitl
Target          Text     Data    BSS     Total  
------------------------------------------------
bin/arducopter  3142821  131427  110496  3384744
```

close https://github.com/ArduPilot/ardupilot/issues/15132

### CODE REVIEW

**Comment by robustini** [comment]
This should also be in the Plane and Rover.

**Comment by khancyr** [comment]
@robustini that is already in rover and plane ;-P

**Comment by khancyr** [comment]
rebased and add some testing to validated that it change nothing on current behavior and that the initial mode is working for all vehicle without radio !

**Comment by muramura** [comment]
I don't think it is necessary to set the initial slow mode because it can be set by user task and mavlink message.  It can be specified at startup with gcs and opentx.  Over 1000, the mission planner is starting very slowly.

**Comment by khancyr** [comment]
There are some value to have the initialising mode.
- It should depend on nothing so we won't break to initial step by making modifications on mode. 
- It doesn't allow arming or anything so so risk of issue or unclear state
- it will allow to make easier diagnostic if an issues appear on initial sequence
- it unifies the initial sequence with other vehicle that also got the initialising mode

**Review by rmackay9 — CHANGES_REQUESTED** [review]
I don't think we actually need the initialising flight mode for this to work though.  Copter always allows changing into any mode while disarmed.

Could you modify the parameter description so that the modes have only the first letter capitalised?  The list of modes seems to come from Plane so let's fix that too. txs

**Comment by khancyr** [comment]
@rmackay9 thank for the input, I have corrected. 
I know that the initilising mode isn't need but this is consistent with other vehicle and provide some advantage that I list in my previous bullet points.

**Comment by EosBandi** [comment]
Just for my 2 cents, In the last two months our pilot managed to takeoff in stabilize mode instead of althold three times, because Herelink uses mavlink to change mode and after reboot mode resets to stabilize. Trust me it is not fun when a 25+Kg hexa plays rocket instead of nice controlled ascend.
Yes, I know that this is pilot error and can be avoided with strict pre-flight checklist, etc. but being able to set initial flight mode would provide a nice additional security.

**Comment by rmackay9** [comment]
@khancyr,

I quickly tested this in SITL and found two problems:

- It is possible to enter initial mode while flying.  I tried this in SITL and experienced a fly away
- I could switch to initial mode and even arm in it if i used "arm throttle force"

I'm really not keen on having the new flight mode when it's not necessary.  One of the golden rules of AP is we don't add things unless they are required.  I probably won't merge this with the initial mode included.

I think another important feature we should consider is how does this work if the pilot has an RC transmitter attached (which most users do).  Are we going to ignore the pilot's channel 5 initial position on startup?  What if the RC is off when the vehicle is powered up but then the signal arrives 2min (or 2 hours) later?

**Comment by khancyr** [comment]
Thanks for the inputs !
I have test this with other vehicle too and we got the same issues (doesn't even need force arm). I will raise an issue and another PR to correct those.
I will remove the initialising mode for now.

About the RC, the behaviour isn't changed, when you connect the radio the mode switch is activated and you switch the mode to the one settled, see in the test. Nothing change on this side. It is just that the copter won't always start on STABILIZE without a radio now.

**Comment by muramura** [comment]
@khancyr san. 
This setting is good if you only support one flight mode.
However, if you want to re-fly after disarming in RTL or LAND, you will have to plug and unplug the battery.
I use SET_MODE to change the flight mode when I want to fly again.
I don't think initial flight mode setting is necessary because I use LUA script and user task to handle it.

File: ArduCopter/Parameters.cpp
Comment (tridge) [inline ArduCopter/Parameters.cpp:314]: where is standby mode? That is likely the most useful

**Comment by rmackay9** [comment]
Merged, thanks!

File: ArduCopter/Parameters.cpp
Comment (khancyr) [reply ArduCopter/Parameters.cpp:314]: what do you call standby mode ?

Comment (rmackay9) [reply ArduCopter/Parameters.cpp:314]: Ah, Standby mode is not actually a flight mode.  It's a modifier of any mode (I think).  For reference here is a list of modes: https://github.com/ArduPilot/ardupilot/blob/master/ArduCopter/mode.h#L15

**Comment by EosBandi** [comment]
Hmm, i see it is merged. Meanwhile I managed to get the same result with six lines of lua. It is called once after bootup, and make sure if the lua engine restart in flight it will not change mode while vehicle is armed.
```

function update()
	if (arming:is_armed() == false) then
		vehicle:set_mode(5) -- Loiter, but you can set your bootup mode here
	end
end
return update, 1000
```

**Comment by rmackay9** [comment]
@EosBandi great that you're putting Lua to good use!

**Pull request merged**
